api/src/main/java/io/strimzi/api/kafka/model/KafkaJmxOptions.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+@EqualsAndHashCode
+public class KafkaJmxOptions implements UnknownPropertyPreserving, Serializable {
+    private static final long serialVersionUID = 1L;
+    private Boolean authentication;
+
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("When `authentication` is enabled, then it will secure the remote JMX port with a username and password.")
```

Comment:
A general question to the APi design. Do we plan to support only one type of authentication in the long time? If you are not sure, maybe we should consider implementing this as:

```
authentication:
  type: password
```

instead of 

```
authentication: true
```

because the first one gives us options to add different types in the future.